build-tools/packages/build-cli/src/commands/check/layers.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "path";
+import { Flags } from '@oclif/core';
+import { LayerGraph, Timer, writeFileAsync } from "@fluidframework/build-tools";
+import { BaseCommand } from "../../base";
+
+// eslint-disable-next-line @typescript-eslint/no-inferrable-types
+const packagesMdFileName: string = "PACKAGES.md";
+
+export class LayerCheck extends BaseCommand {
+  static description = 'Make sure the dependencies between Fluid Framework packages are properly layered';
+
+  static flags = {
+    md: Flags.string({
+        description: "Generate PACKAGES.md file for human consumption at path relative to repo root",
+        required: false,
+        default: ".", // default is repo root (relative path to repo root)
+     }),
+    dot: Flags.file({
+        description: "Generate *.dot for GraphViz",
+        required: false
+     }),
+    info: Flags.string({
```

Comment:
```suggestion
    info: Flags.file({
```

I think this gets us additional argument validation for free.